IdfaIOS/CustomIdfa.m

```diff
@@ -0,0 +1,113 @@
+#import <AdSupport/AdSupport.h>
+#import <AppTrackingTransparency/AppTrackingTransparency.h>
+#import <CoreLocation/CoreLocation.h>
+
+//Please ensure this path points to the file 'Assets/Plugins/iOS/Amplitude/Amplitude.h'
+#import "../../Plugins/iOS/Amplitude/Amplitude.h"
+
+typedef NSString* (^AMPAdSupportBlock)(void);
+typedef NSDictionary* (^AMPLocationInfoBlock)(void);
+
+// Converts C style string to NSString
+// Note, this will produce C++ compiler/linker errors if this function is renamed to ToNSString or any function in Objective C++ files
+NSString* ToNSString2(const char* string)
+{
+    if (string)
+        return [NSString stringWithUTF8String: string];
+    else
+        return nil;
+}
```

Comment:
Looks like repeated code in [`AmplitudeCWrapper.m`](https://github.com/amplitude/unity-plugin/blob/a6cff8dcb38d6f3f6ce50708072a5f0a70a3f483/Assets/Plugins/iOS/Amplitude/AmplitudeCWrapper.m#L20-L27), could we use this one instead?

Reply:
I used a new function declared here. I tried to import the header `#import "../../Plugins/iOS/Amplitude/AmplitudeCWrapper.h", but for some reason the xcode linker gives the error 

`error: implicit declaration of function 'ToNSString' is invalid in C99`
implying the function is not accessible